fdbserver/workloads/RemoveServersSafely.actor.cpp

```diff
@@ -401,11 +403,48 @@ struct RemoveServersSafelyWorkload : TestWorkload {
 
 		state std::vector<ISimulator::ProcessInfo*>	killProcArray;
 		state std::vector<AddressExclusion>	toKillArray;
+		state std::vector<AddressExclusion>	toKillMarkFailedArray;
 
 		std::copy(toKill.begin(), toKill.end(), std::back_inserter(toKillArray));
 		killProcArray = self->getProcesses(toKill);
+		if (safeKillSet) {
+			state int timeouts = 0;
+			loop {
+				state bool safe = false;
+				auto failSet = random_subset(toKillArray, deterministicRandom()->randomInt(0, toKillArray.size() + 1));
+				toKillMarkFailedArray.resize(failSet.size());
+				std::copy(failSet.begin(), failSet.end(), toKillMarkFailedArray.begin());
+				TraceEvent("RemoveAndKill", functionId)
+				    .detail("Step", "SafetyCheck")
+				    .detail("Exclusions", describe(toKillMarkFailedArray));
+				loop {
+					choose {
+						when(bool _safe = wait(checkSafeExclusions(cx, toKillMarkFailedArray))) {
+							safe = _safe;
+							break;
+						}
+						when(wait(delay(5.0))) {
+							TraceEvent("RemoveAndKill", functionId)
+							    .detail("Step", "SafetyCheckTimedOut")
+							    .detail("Exclusions", describe(toKillMarkFailedArray));
+						}
+					}
+					if (timeouts == self->maxSafetyCheckTimeouts) {
```

Comment:
Forcibly remove a server may break the DB. I guess the `maxSafetyCheckTimeouts ` value is set too big so that this case is not fully tested.

We can buggify this situation: when it is buggified, we forcibly mark as server as failed. 
To avoid false positive that brings down all replicas, we can only do this for 1 SS when there are more than 1 SS to kill legally.

Reply:
Actually, this force should never break the DB. I confirmed with Evan that the second set of excludes (which is where these servers are marked as failed) will always be safe and not bring down the cluster. This `force` is mostly to keep the test from looping too many times if the safety check request continues to time out.

Reply:
In conjunction with the code review suggestions in removing minReplication=2 and removing a coordinator if BUGGIFY was on, this force was indeed causing broken DBs in certain rare configurations. I've changed the workload to avoid this if the retry limit is reached.